cli/src/common/imagebuilder_utils.py

```diff
@@ -32,6 +35,44 @@ def get_info_for_ami_from_arn(image_arn):
     return AWSApi.instance().imagebuilder.get_image_resources(image_arn)
 
 
+def wrap_script_to_component(url):
+    """Wrap script to custom component data property."""
+    bucket_info = get_bucket_object_name(url)
+    scheme = bucket_info.get("scheme")
+    script_name = bucket_info.get("object_name")
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+
+    if scheme == "https":
+        custom_component_script_template_file = os.path.join(
+            current_dir, "..", "pcluster", "resources", "imagebuilder", "custom_component_script_https.yaml"
+        )
+    else:
+        custom_component_script_template_file = os.path.join(
+            current_dir, "..", "pcluster", "resources", "imagebuilder", "custom_component_script_s3.yaml"
+        )
+
+    with open(custom_component_script_template_file, "r") as file:
+        custom_component_script_template = yaml.load(file)
+
+    script_name_action = custom_component_script_template["phases"][0]["steps"][1]["inputs"]["commands"][0]
```

Comment:
this relies on the ordering of the steps, which could be different if we change the component definition. To make this more robust I see three alternatives:
1- search for the `build` phase and then we search for the action name, e.g.
```
build_step=[item for item in custom_component_script_template["phases"] if item.get("name")=="build"][0]
action_script_name[item for item in build_step["steps"] if item.get("name")=="ScriptName"][0]
```
handling the case when the item is not found

2- read the whole file as a text and just replace the placeholders

3- build a `CfnParameter` only when custom component script is specified and use it inside the `yaml` definition 

Between the three I'm towards to use the CfnParameter

Reply:
Here met some issues when addressing option 1 and 3. 
For option 1, although we are able to find the build_step and action by searching the name, but how can we assign it to the template, we still need to get block by index in template and assign it with replaced string. 
For option3, since we supports multiple scripts, how to use cfnparameter to pass multiple urls.